src/components/Header.tsx

```diff
@@ -0,0 +1,126 @@
+import CloseIcon from "@mui/icons-material/Close";
+import MenuOutlinedIcon from "@mui/icons-material/MenuOutlined";
+import {
+  AppBar,
+  Box,
+  Container,
+  Drawer,
+  IconButton,
+  Toolbar,
+} from "@mui/material";
+import { useEffect, useState } from "react";
+import { useLocation } from "react-router-dom";
+import AuthButtons from "./AuthButtons";
+import Logo from "./Logo";
+import NavItems from "./NavItems";
+
+const Header = () => {
+  const [open, setOpen] = useState<boolean>(false);
+  const [scrolled, setScrolled] = useState<boolean>(false);
+  const location = useLocation();
+
+  const toggleDrawer = (newOpen: boolean) => () => {
+    setOpen(newOpen);
+  };
+
+  useEffect(() => {
+    const handleScroll = () => {
```

Comment:
maybe this handle scroll could be removed out of useEffect. 

useEffect(()=>{

handleScroll()
},[])